crates/rpc/eth/logs.rs

```diff
@@ -0,0 +1,153 @@
+    pub toBlock: BlockIdentifier,
+    /// The addresses from where the logs origin from.
+    pub address: Option<Vec<H160>>,
+    /// Which topics to filter.
+    pub topics: Option<Vec<U256>>,
+}
+impl RpcHandler for LogsRequest {
+    fn parse(params: &Option<Vec<Value>>) -> Result<LogsRequest, RpcErr> {
+        match params.as_deref() {
+            Some([param]) => {
+                let param = param.as_object().ok_or(RpcErr::BadParams)?;
+                let fromBlock = {
+                    if let Some(param) = param.get("fromBlock") {
+                        BlockIdentifier::parse(param.clone(), 0)?
+                    } else {
+                        BlockIdentifier::latest()
+                    }
+                };
+                let toBlock = {
+                    if let Some(param) = param.get("toBlock") {
+                        BlockIdentifier::parse(param.clone(), 1)?
+                    } else {
+                        BlockIdentifier::latest()
+                    }
+                };
+                let address = None;
+                let topics = None;
+                Ok(LogsRequest {
+                    fromBlock,
+                    address,
+                    topics,
+                    toBlock,
+                })
+            }
+            Some(params) => unreachable!("{params:?}"),
+            _ => Err(RpcErr::BadParams),
+        }
+    }
+    // TODO: This is longer than it has the right to be, we should refactor it.
+    // The main problem here is the layers of indirection needed
+    // to fetch tx and block data for a log rpc response, some ideas here are:
+    // - The ideal one is to have a key-value store BlockNumber -> Log, where the log also stores
+    //   the block hash, transaction hash, transaction number and its own index.
+    // - Another on is the receipt stores the block hash, transaction hash and block number,
+    //   then we simply could retrieve each log from the receipt and add the info
+    //   needed for the RPCLog struct.
+    fn handle(&self, storage: Store) -> Result<Value, RpcErr> {
+        let Ok(Some(from)) = self.fromBlock.resolve_block_number(&storage) else {
+            return Err(RpcErr::BadParams);
+        };
+        let Ok(Some(to)) = self.toBlock.resolve_block_number(&storage) else {
+            return Err(RpcErr::BadParams);
+        };
+        let receipts = storage.get_receipts_in_range(from, to)?;
+
+        // To build the RPC response, we'll need some information besides the log,
+        // these are:
+        // - Each tx hash + index inside the block
+        // - Each Block Number (and eventually, hash)
+        // This is what each triple of this vec represents:
+        // - First coordinate is hash of a tx.
+        // - Second coordinate is the index inside the tx.
+        // - Third coordinate is the block number on which this tx was executed.
+        let tx_and_block_data: Vec<(H256, Index, BlockNumber)> = receipts
```

Comment:
why don't you query block bodies to get the transactions instead?

Reply:
Yup, I took that approach.